### DISCUSSION TITLE
How to fix WinPE screen resolution

### DISCUSSION THREAD

**Discussion by skyblaster** [ipxe/wimboot#58] [Q&A] [answered]
When I boot a WinPE boot.wim from USB, the screen resolution looks crisp with the proper aspect ratio.
When I boot the same wim along with the accompanying bcd and efi files, WinPE ends up looking stretched horizontally.

Here is the script I'm currently using to troubleshoot. The only line that seems to present any changes is the "console --x 1920 --y 1080" line. This simply accounts for smaller console text before the WIM is loaded.
The other resolution based lines are commented because I couldn't see any change when uncommented.

```
#!ipxe
dhcp net0
console --x 1920 --y 1080
#set video-output resolution=1920x1080
#set resolution 1920 1080
#set mode 1920 1080
set gateway 0.0.0.0
set boot-url http://${dhcp-server}
  kernel ${boot-url}/wimboot
  initrd ${boot-url}/BOOTMGR.EFI          BOOTMGR.EFI
  initrd ${boot-url}/BOOT/BCD             BCD
  initrd ${boot-url}/BOOT/BOOT.SDI        BOOT.SDI
  initrd ${boot-url}/SOURCES/X64/BOOT.WIM BOOT.WIM
  imgstat
  prompt
  boot
```

As suggested elsewhere online, I tried adding the following unattend.xml to my image, however there was no change.
```
<Display>
  <HorizontalResolution>1920</HorizontalResolution>
  <VerticalResolution>1080</VerticalResolution>
  <ColorDepth>32</ColorDepth>
  <RefreshRate>60</RefreshRate>
</Display>
```

Hopefully there is a simple fix that I'm overlooking.

**Comment by NiKiZe** [comment, marked as answer]
Test adding `gui` to your kernel line.

Also if you don't have a bored boot.wim all the other files will be loaded from the wim and not required to be loaded separately.

**Reply by skyblaster** [reply]
> Test adding `gui` to your kernel line.

That did the trick. Thank you!

> Also if you don't have a bored boot.wim all the other files will be loaded from the wim and not required to be loaded separately.

Here is my new minimal config. I did not have success booting unless I added the bcd and boot.sdi lines back in.
```
#!ipxe
dhcp net0
console --x 1920 --y 1080
set gateway 0.0.0.0
set boot-url http://${dhcp-server}
  kernel ${boot-url}/wimboot gui
  initrd ${boot-url}/BOOT/BCD             BCD
  initrd ${boot-url}/BOOT/BOOT.SDI        BOOT.SDI
  initrd ${boot-url}/SOURCES/X64/BOOT.WIM BOOT.WIM
  boot
```

**Comment by skyblaster** [comment]
Here are my best screen captures before final load:
![image](https://github.com/user-attachments/assets/18828bb6-141d-49b9-92e1-298b8bf394cc)

![image](https://github.com/user-attachments/assets/d1403630-7a47-463a-a312-c6d3bdcb2f23)

**Reply by skyblaster** [reply]
Ah, yes. I mounted my boot.wim and sure enough bcd nor boot.sdi are present.
They are only contained in the ISO that the WIM was extracted from.....so I guess they have to be loaded via TFTP or HTTP.

Thanks again!
